### DISCUSSION TITLE
Edge Functions: AuthUnknownError in initializePromise

### DISCUSSION THREAD

**Discussion by lukas-stein** [supabase/supabase#22319] [Questions]
Hi everyone,

I am writing a supabase edge function and for some unknown reason my client throws an error after initialisation.

This is my code with the error in comments:

```ts
  const authHeader = req.headers.get("Authorization")!;
  const supabaseClient = createClient(
    Deno.env.get("SUPABASE_URL") ?? "",
    Deno.env.get("SUPABASE_ANON_KEY") ?? "",
    { global: { headers: { Authorization: authHeader } } },
  );

  // -----------------------

  console.log('Client 0:', supabaseClient) // initializePromise: Promise { <pending> },

  // Get the user object
  const { data } = await supabaseClient.auth.getUser()
  const user = data.user

  console.log('Client 1:', supabaseClient) // initializePromise: Promise {{error: AuthUnknownError: Unexpected error during initialization...
```

I am not sure if the getUser function has anything to do with it but the error consistently occurs after. However, the user object itself is accurate.

Does anybody have a solution?

Thanks.

**Comment by GaryAustin1** [comment]
You can't use getUser() without a token when you "force" the authorization header.  There is no session in the client for it to pickup.
You could extract the user jwt from the authorization header and do .getUser(jwt) and that should work.

**Reply by GaryAustin1** [reply]
https://github.com/orgs/supabase/discussions/19621 has a user's example of getting the jwt.

**Reply by lukas-stein** [reply]
Thanks! Interestingly, I have no trouble getting the user object. It's just screwing with my client. 

I tried doing getUser(jwt) but the client still shows an initialization error after.

Or am I misunderstanding something?

**Reply by GaryAustin1** [reply]
You are getting an error on the getUser() call?  Why don't you get and console.log the error returned on that to see if that is the error.

**Reply by lukas-stein** [reply]
No, sorry for not being clear about that.

The issue is with my supabaseClient. After calling the getUser function, an error appears in the auth header of the client when I log the client in the console:

```
Client 1: f {
  supabaseUrl: [...],
  supabaseKey: [...]... 108 more characters,
  // ...
  headers: {
    Authorization: "Bearer [...]
  },
  auth: _ {
    memoryStorage: {},
    stateChangeEmitters: Map(1) {
      "[...]" => {
        id: "[...]",
        callback: [Function (anonymous)],
        unsubscribe: [Function: unsubscribe]
      }
    },
    autoRefreshTicker: 1,
    visibilityChangedCallback: null,
    refreshingDeferred: null,
    initializePromise: Promise {
      {
        error: AuthUnknownError: Unexpected error during initialization
    at _._initialize [...]
```

The actual problem I am having is calling the updateUser function after, as it throws an `AuthSessionMissingError`. I just suspect it has something to do with the error in the auth.

**Reply by GaryAustin1** [reply]
That is because there is no real auth session when you set the authorization header.  You are not logged in as the user, just have the ability to run database/storage operations as them.
